lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1210,11 +1216,41 @@ LLVMToSPIRVBase::getLoopControl(const BranchInst *Branch,
           BM->addExtension(ExtensionID::SPV_INTEL_fpga_loop_controls);
           BM->addCapability(CapabilityFPGALoopControlsINTEL);
           LoopControl |= spv::LoopControlNoFusionINTELMask;
+        } else if (S == "llvm.loop.intel.loopcount_min") {
+          BM->addExtension(ExtensionID::SPV_INTEL_fpga_loop_controls);
+          BM->addCapability(CapabilityFPGALoopControlsINTEL);
+          LoopCountMin = getMDOperandAsInt(Node, 1);
+          LoopControl |= spv::internal::LoopControlLoopCountINTELMask;
+        } else if (S == "llvm.loop.intel.loopcount_max") {
+          BM->addExtension(ExtensionID::SPV_INTEL_fpga_loop_controls);
+          BM->addCapability(CapabilityFPGALoopControlsINTEL);
+          LoopCountMax = getMDOperandAsInt(Node, 1);
+          LoopControl |= spv::internal::LoopControlLoopCountINTELMask;
+        } else if (S == "llvm.loop.intel.loopcount_avg") {
+          BM->addExtension(ExtensionID::SPV_INTEL_fpga_loop_controls);
+          BM->addCapability(CapabilityFPGALoopControlsINTEL);
+          LoopCountAvg = getMDOperandAsInt(Node, 1);
+          LoopControl |= spv::internal::LoopControlLoopCountINTELMask;
         }
       }
     }
   }
-
+  if (LoopControl & spv::internal::LoopControlLoopCountINTELMask) {
+    // LoopCountINTELMask have int64 literal parameters and we need to store
+    // int64 into 2 SPIRVWords
+    ParametersToSort.emplace_back(spv::internal::LoopControlLoopCountINTELMask,
```

Comment:
From description of std::sort: "The order of equal elements is not guaranteed to be preserved."
Maybe I'm just getting worked up over nothing, but may there be a situation when second word of one of these params will be moved before first word or even before a word of another parameter?

Reply:
> From description of std::sort: "The order of equal elements is not guaranteed to be preserved."

Unless a programmer wrote a special rule how to proceed with the sort in a lambda, and here we have:
`return CompareLeft.first < CompareRight.first`

Reply:
I meant that some sort algorithms are unstable and may change order of equal elements even for strict less comparison.

Reply:
Oh, I found that there is also present std::stable_sort. Maybe to use this sort to be sure that there won't be unexpected reshuffles of LoopCountXXX words?

Reply:
> Maybe to use this sort to be sure that there won't be unexpected reshuffles of LoopCountXXX words?

Do we actually care about those re-orders?

Reply:
Changed std::sort to std::stable_sort in [1fbf78b](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1062/commits/1fbf78bf40440051bb7dbc9459d4cb1070ea8eae)